src/coreclr/jit/lower.cpp

```diff
@@ -9339,6 +9405,22 @@ bool Lowering::TryMakeIndirsAdjacent(GenTreeIndir* prevIndir, GenTreeIndir* indi
         node = next;
     }
 
+    if (sawData)
+    {
+        // For some reason LSRA is not able to reuse a constant if both LIR
+        // temps are live simultaneously, so skip moving in those cases and
+        // expect LSRA to reuse the constant instead.
+        if (!indir->Data()->IsCnsIntOrI() || !GenTree::Compare(indir->Data(), prevIndir->Data()))
```

Comment:
should it make an exception for 0 value? since ZR reg shouldn't be a problem for LSRA

Reply:
For 0 the constant will be contained in the `STOREIND`, so whether we move it or not shouldn't matter since it won't result in any instructions generated between the two indirs regardless.